bitcoin/src/address.rs

```diff
@@ -1057,6 +1057,13 @@ impl Address<NetworkUnchecked> {
     pub fn assume_checked(self) -> Address { Address::new(self.network, self.payload) }
 }
 
+// For NetworkUnchecked , it compare Addresses and if network and payload matches then return true.
+impl PartialEq<Address<NetworkUnchecked>> for Address {
+    fn eq(&self, other: &Address<NetworkUnchecked>) -> bool {
+        self.network == other.network && self.payload == other.payload
+    }
+}
```

Comment:
This only implements `checked == unchecked`, but `unchecked == checked` still wouldn't work. You need to also implement `PartialEq<Address> for Address<NetworkUnchecked>`. (You can just call `other == self` there.)

Reply:
Sure. Thanks for guiding. I have implemented this as well.